core/src/main/java/org/mule/runtime/core/policy/SourcePolicy.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.policy;
+
+import org.mule.runtime.core.api.Event;
+
+import java.io.Serializable;
+
+/**
+ * A {@code SourcePolicy} is responsible to handle the state of a policy applied to a particular execution.
+ *
+ * @since 4.0
+ */
+public interface SourcePolicy extends Serializable {
```

Comment:
sounds like this and many other interfaces in this commit should be API. maybe not Mule API, but core module API at the very least